conftest.py

```diff
@@ -455,6 +472,13 @@ def environment_cleanup(monkeypatch: pytest.MonkeyPatch) -> None:
     monkeypatch.delenv("SSKEYOUT", raising=False)
 
 
+@pytest.fixture(scope="function", autouse=True)
+def check_output_dir() -> None:
+    global test_output_dirs
+    assert os.path.isdir(test_output_root)
+    assert len(os.listdir(test_output_root)) >= test_output_dirs
+    test_output_dirs = len(os.listdir(test_output_root))
```

Comment:
Super small nit: this will only list for dirs/files/etc. `tests/test_output` _specifically_ (i.e. it will ignore files/dirs/etc. in `tests_output/backends/*` for e.g.).

Is this intended? If not, I would recommend switching to `os.walk` and caching the value for these two length checks (it should still be pretty fast bc we are not writing to the fs, but there are now significantly more files to iterate over)
```suggestion
    files_and_dirs = itertools.chain.from_iterable(
        (fs, ds) for _, fs, ds, in os.walk(test_output_root)
    )
    num_files_and_dirs = len(tuple(itertools.chain.from_iterable(files_and_dirs)))
    assert num_files_and_dirs >= test_output_dirs
    test_output_dirs = num_files_and_dirs
```
If this is intended, feel free to disregard!

Reply:
Yeah, my idea was just to check first level, in case something bad happened that removed all dirs in `test_output`, either b/c of a faulty test, or because of filesystem issues. I am just afraid that running `os.walk` in every test could lead to way longer testing times

Reply:
Makes perfect sense to me! Totally agree with the worry of `os.walk` taking too long, and as long as this limitation is known, I see no problem with this solution!!